WordPress/gutenberg
# URL: Fix code drift in the Editor package by removing duplicate cleanForSlug function

I was investigating reports that the object replacement character could be added to a post slug, and noticed that we had two definitions of `cleanForSlug`, with diverging behavior.

I fixed this by updating the function in the url package to include fixes from https://github.com/WordPress/gutenberg/issues/24710, and re-exporting the function in the editor package. I think we can probably deprecate the url utils in the editor package (in either a follow up or in this PR), but let me know if folks have opinions on this.

Note that this is client-side sanitization, so this will help prevent new bad entries. For posts/slugs with existing bad characters we'll need a separate patch for server sanitization.

cc @claudiulodro @Robertght would it be possible to see if this also helps with: https://github.com/WordPress/gutenberg/issues/38637

### Testing Instructions

In console:
```js
wp.url.cleanForSlug('Καλημέρα Κόσμε');
```
returns `'καλημέρα-κόσμε'` instead of an empty string.

```js
wp.editor.cleanForSlug('Καλημέρα Κόσμε');
```
Has the same result, but also includes a deprecation warning.

<img width="716" alt="CleanShot 2022-02-23 at 10 41 12@2x" src="https://user-images.githubusercontent.com/1270189/155386100-989d17a8-d09d-46bc-a455-de539b03358f.png">


- In the post editor, make sure permalinks are set to include the post title
- Create a new post and save a draft
- Try editing the permalink slug to include non-latin characters, and unicode characters like the object replacement characters
- On blur unexpected characters should be sanitized.
